pennylane/qchem/factorization.py

```diff
@@ -169,3 +170,145 @@ def factorize(two_electron, tol_factor=1.0e-5, tol_eigval=1.0e-5):
+     array([-1.36155423,  2.03646071, -1.34981296,  0.67490648])]
+
+    .. details::
+        :title: Theory
+
+        A second-quantized molecular Hamiltonian can be constructed in the
+        `chemist notation <http://vergil.chemistry.gatech.edu/notes/permsymm/permsymm.pdf>`_ format
+        following Eq. (1) of
+        [`PRX Quantum 2, 030305, 2021 <https://journals.aps.org/prxquantum/abstract/10.1103/PRXQuantum.2.030305>`_]
+        as
+
+        .. math::
+
+            H = \sum_{\alpha \in \{\uparrow, \downarrow \} } \sum_{pq} T_{pq} a_{p,\alpha}^{\dagger}
+            a_{q, \alpha} + \frac{1}{2} \sum_{\alpha, \beta \in \{\uparrow, \downarrow \} } \sum_{pqrs}
+            V_{pqrs} a_{p, \alpha}^{\dagger} a_{q, \alpha} a_{r, \beta}^{\dagger} a_{s, \beta},
+
+        where :math:`V_{pqrs}` denotes a two-electron integral in the chemist notation and
+        :math:`T_{pq}` is obtained from the one- and two electron integrals, :math:`h_{pq}` and
+        :math:`h_{pssq}`, as
+
+        .. math::
+
+            T_{pq} = h_{pq} - \frac{1}{2} \sum_s h_{pssq}.
+
+        The tensor :math:`V` can be converted to a matrix which is indexed by the indices :math:`pq`
+        and :math:`rs` and eigendecomposed up to a rank :math:`R` to give
+
+        .. math::
+
+            V_{pqrs} = \sum_r^R L_{pq}^{(r)} L_{rs}^{(r) T},
+
+        where :math:`L` denotes the eigenvectors of the matrix. The molecular Hamiltonian can then
+        be rewritten following Eq. (7) of
+        [`Phys. Rev. Research 3, 033055, 2021 <https://journals.aps.org/prresearch/abstract/10.1103/PhysRevResearch.3.033055>`_]
+        as
+
+        .. math::
+
+            H = \sum_{\alpha \in \{\uparrow, \downarrow \} } \sum_{pq} T_{pq} a_{p,\alpha}^{\dagger}
+            a_{q, \alpha} + \frac{1}{2} \sum_r^R \left ( \sum_{\alpha, \beta \in \{\uparrow, \downarrow \} } \sum_{pq}
+            L_{pq}^{(r)} a_{p, \alpha}^{\dagger} a_{q, \alpha} \right )^2.
+
+        The orbital basis can be rotated such that each :math:`T` and :math:`L^{(r)}` matrix is
+        diagonal. The Hamiltonian can then be written following Eq. (2) of
+        [`npj Quantum Information, 7, 23 (2021) <https://www.nature.com/articles/s41534-020-00341-7>`_]
+        as
+
+        .. math::
+
+            H = U_0 \left ( \sum_p d_p n_p \right ) U_0^{\dagger} + \sum_r^R U_r \left ( \sum_{pq}
+            d_{pq}^{(r)} n_p n_q \right ) U_r^{\dagger}
+
+        where the coefficients :math:`d` are obtained by diagonalizing the :math:`T` and
+        :math:`L^{(r)}` matrices. The number operators :math:`n_p = a_p^{\dagger} a_p` can be
+        converted to qubit operators using
+
+        .. math::
+
+            n_p = \frac{1-Z_p}{2}
+
+        where :math:`Z_p` is the Pauli :math:`Z` operator applied to qubit :math:`p`. This gives
+        the qubit Hamiltonian
+
+        .. math::
+
+           H = U_0 \left ( \sum_p O_p^{(0)} \right ) U_0^{\dagger} + \sum_r^R U_r \left ( \sum_{q} O_q^{(r)} \right ) U_r^{\dagger},
```

Comment:
What is meant by `p^{(0)}` and `q^{(r)}`?

Reply:
These are `O_p^{(0)}` and `O_q^{(r)}`, where O is defined in the line below the equation as `O = \sum_i c_i P_i`. The labels `0` and `r` are added for consistency with the unitaries `U_0` and `U_r`. Since `r` starts from `1`, we have used `0` for the first unitary, following the original paper.